client/setec/store.go

```diff
@@ -418,7 +479,22 @@ func (s *Store) applyUpdates(updates map[string]*api.SecretValue) error {
 	s.active.Lock()
 	defer s.active.Unlock()
 	for name, sv := range updates {
-		s.active.m[name] = sv
+		if sv == nil {
+			// This is an undeclared secret that has expired.
+			// If there are no handles referring to it, remove it.
+			if _, ok := s.active.f[name]; ok {
+				// This secret has an outstanding handle. Since watchers package
```

Comment:
How can it have an outstanding handle but be inactive?

Reply:
Say, the service starts up and someone grabs an undeclared secret. The service runs for several weeks, during which time the caller doesn't actually access the secret, so it expires. Now we go to expire it.

It's not a super likely situation, but I didn't want it to fail broken.

Reply:
(added an explanatory comment)